lib/widgets/resources/arm.dart

```diff
@@ -0,0 +1,192 @@
+  }
+
+  Future<void> updateJointPosition(int joint, double increment) async {
+    final jp = jointPositions;
+    jp[joint] += increment;
+    await widget.arm.moveToJointPositions(jp);
+    await _getPositions();
+  }
+
+  TableRow _getEndPositionRow(_PoseField field) {
+    double value;
+    switch (field) {
+      case _PoseField.x:
+        value = endPosition.x;
+      case _PoseField.y:
+        value = endPosition.y;
+      case _PoseField.z:
+        value = endPosition.z;
+      case _PoseField.theta:
+        value = endPosition.theta;
+      case _PoseField.oX:
+        value = endPosition.oX;
+      case _PoseField.oY:
+        value = endPosition.oY;
+      case _PoseField.oZ:
+        value = endPosition.oZ;
+    }
+
+    return TableRow(children: [
+      TableCell(child: Padding(padding: const EdgeInsets.symmetric(horizontal: 2), child: Text(field.title, textAlign: TextAlign.end))),
+      TableCell(
+          child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 2),
+              child: ViamButton(onPressed: () => updateEndPosition(field, -10), text: '--', size: ViamButtonSizeClass.small))),
+      TableCell(
+          child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 2),
+              child: ViamButton(onPressed: () => updateEndPosition(field, -1), text: '-', size: ViamButtonSizeClass.small))),
+      TableCell(
+          child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 2), child: Text(value.toStringAsFixed(2), textAlign: TextAlign.center))),
+      TableCell(
+          child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 2),
+              child: ViamButton(onPressed: () => updateEndPosition(field, 1), text: '+', size: ViamButtonSizeClass.small))),
+      TableCell(
+          child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 2, vertical: 1),
+              child: ViamButton(onPressed: () => updateEndPosition(field, 10), text: '++', size: ViamButtonSizeClass.small))),
+    ]);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: [
+        const Text('End Positions (mm)', style: TextStyle(fontWeight: FontWeight.bold)),
+        Table(
+          columnWidths: const {0: IntrinsicColumnWidth()},
+          defaultVerticalAlignment: TableCellVerticalAlignment.middle,
+          children: _PoseField.values.map((e) => _getEndPositionRow(e)).toList(),
+        ),
+        const SizedBox(height: 16),
+        const Text('Joints (degrees)', style: TextStyle(fontWeight: FontWeight.bold)),
+        Table(
+          columnWidths: const {0: IntrinsicColumnWidth()},
+          defaultVerticalAlignment: TableCellVerticalAlignment.middle,
+          children: jointPositions
+              .mapIndexed((index, element) => TableRow(children: [
```

Comment:
I think this `TableRow` should still be broken out into it's own widget as well.